src/app/services/lastSearches.js

```diff
@@ -0,0 +1,27 @@
+export default class LastSearchesService {
+    constructor(lastSearches) {
+        this.lastSearches = lastSearches;
+    }
+
+    saveCity(city) {
+        if (city === undefined || city.trim() === '') {
+            return;
+        }
+        const cityAux = city.trim().toLowerCase();
+        if (this.lastSearches.includes(cityAux)) {
+            return;
+        }
+        this.lastSearches.unshift(cityAux);
+        if (this.lastSearches.length > 10) {
+            this.lastSearches.pop();
```

Comment:
Since the condition is a length bigger than 10, maybe is a good idea to get only the first 10 elements, instead of remove the last